src/components/AppShell.jsx

```diff
@@ -0,0 +1,129 @@
+import {
+  AppShell,
+  Burger,
+  Footer,
+  Header,
+  MediaQuery,
+  Menu,
+  Navbar,
+  Stack,
+  Text,
+  UnstyledButton,
+  useMantineTheme,
+} from "@mantine/core";
+import {
+  IconLogout,
+  IconMoon,
+  IconSettings,
+  IconSunFilled,
+  IconSwitchHorizontal,
+  IconUserCircle,
+} from "@tabler/icons-react";
+import { useState } from "react";
+import { DemoComponent } from "./DemoComponent/DemoComponent";
+
+export default function AppShellDemo() {
+  const theme = useMantineTheme();
+  const [opened, setOpened] = useState(false);
```

Comment:
As the value `opened` is a boolean value, it's better to name it like `isOpened`

Reply:
Make sence!